test/hotspot/jtreg/runtime/Monitor/StressWrapper_TestRecursiveLocking_36M.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test id=Xint_outer_inner
+ * @requires vm.flagless
+ * @summary Tests recursive locking in -Xint in outer then inner mode.
+ * @library /testlibrary /test/lib /runtime/Monitor
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ *
+ * @run main/othervm/timeout=240 -Xbootclasspath/a:.
+ *     -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI
+ *     -Xint
+ *     -XX:LockingMode=0
+ *     -ms256m -mx256m
+ *     TestRecursiveLocking 120 1
+ *
+ * @run main/othervm/timeout=240 -Xbootclasspath/a:.
+ *     -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI
+ *     -Xint
+ *     -XX:LockingMode=1
+ *     -ms256m -mx256m
+ *     TestRecursiveLocking 120 1
+ *
+ * @run main/othervm/timeout=240 -Xbootclasspath/a:.
+ *     -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI
+ *     -Xint
+ *     -XX:LockingMode=2
+ *     -ms256m -mx256m
+ *     TestRecursiveLocking 120 1
+ */
+
+/*
+ * @test id=Xint_alternate_AB
```

Comment:
I have a question to Misha (@ can't find name) to ask if these are in different files, will the tests run in parallel?  Or does the separate @ test's do that.

Reply:
AFAIK the files should not matter in this case, the @test matters. Got same info from the quality team.

Reply:
Each comment block that begins with an `@test` will execute each `@run` in
the comment block serially.

Each separate `@test` block will run in parallel if JOBS is > 1 and there are
resources available for parallel testing. There are a total of 6 separate `@test`
comment blocks so it's possible that there will be 6 parallel executions.

Each of the 6 separate `@test` comment blocks contains 3 `@run` commands
so each of the 6 `@test` comment blocks will serially execute the 3 `@run`
commands in their own comment block.

If you put all of the `@test` directives in a single comment block, then all of the
`@run` commands will execute serially. However, in this case, it won't work right
because each of the 6 `@test` comment blocks contains various `@requires`
directives that would conflict with each other if they were all in the same comment
block.

Reply:
Ok, good.